drivers/include/drivers/interfaces/InterfaceDigitalOut.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Mbed-OS Microcontroller Library
+ * Copyright (c) 2021 Embedded Planet
+ * Copyright (c) 2021 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
+ */
+
+#ifndef MBED_INTERFACE_DIGITALOUT_H_
+#define MBED_INTERFACE_DIGITALOUT_H_
+
+#include "PinNames.h"
+
+namespace mbed {
+
+class DigitalOut;
+
+namespace interface {
+
+#ifdef FEATURE_EXPERIMENTAL_API
+
+// TODO - move method doxygen comments to interface once this polymorphism is mainstream
```

Comment:
Can we remove this todo ,as this is almost ready for integration,it will go into the next release . If there is anything outstanding, create an issue about it.